### PR TITLE
Update Ruby documentation to reflect current state

### PR DESCRIPTION
**Pull request by vinistock**
The Ruby LSP project has evolved significantly since we added the first iteration of the documentation. This PR updates the docs to reflect the current state of features.

The summary of the changes are:

1. Included a section on navigation and intellisense, which we've been putting a lot of effort into. I added a video to showcase all features at a glance. Please let me know if the size/format is okay. I based it on the videos I found for the JavaScript docs
2. Updated the format of the version manager configuration, which we changed
3. Removed the code snippets section. It honestly doesn't add much value and we're planning on moving all of that to the server eventually so that completion is more accurate
4. Updated the debugging section to mention that the Ruby LSP now has a built-in debugger client. The client supports both launch configurations and usage through code lenses
5. Finally, we originally had two repositories for the VS Code extension and server (`Shopify/vscode-ruby-lsp` and `Shopify/ruby-lsp`). The separation was causing a lot of confusion, so we merged everything into a monorepo in `Shopify/ruby-lsp`. I updated all references to the old repo

### CODE REVIEW

File: docs/languages/images/ruby/navigation.mp4
Comment (ntrogh) [inline docs/languages/images/ruby/navigation.mp4:1]: Nit: could you make sure the video doesn't have the top of the screen cut off?

File: docs/languages/ruby.md [hunk 1]
```diff
@@ -44,20 +44,27 @@ The extension generates a `.ruby-lsp` folder automatically with a custom bundle
...
-If you have any problems, see [troubleshooting](https://github.com/Shopify/vscode-ruby-lsp#troubleshooting) for next steps.
+If you have any problems, see [troubleshooting](https://github.com/Shopify/ruby-lsp/blob/main/TROUBLESHOOTING.md) for next steps.
 
 ## Main features
 
-### Symbol detection
+### Navigation and IntelliSense
 
-Ruby LSP parses source code and provides symbol detection and navigation for the Outline pane, file breadcrumbs, and symbol searches (accessible with `kb(workbench.action.gotoSymbol)` by default).
+The Ruby LSP provides several navigation and IntelliSense related features such as go to definition, hover, workspace
```
Comment (ntrogh) [inline docs/languages/ruby.md:62]:
```suggestion
The Ruby LSP provides several navigation and IntelliSense related features, such as go to definition, hover, workspace
```

File: docs/languages/ruby.md [hunk 2]
```diff
@@ -44,20 +44,27 @@ The extension generates a `.ruby-lsp` folder automatically with a custom bundle
...
+symbol, document symbol, completion and signature help.
 
-![Ruby document symbols](images/ruby/ruby_lsp_symbols.png)
+<video src="images/ruby/navigation.mp4" placeholder="images/ruby/navigation-placeholder.png" autoplay loop controls muted>
```
Comment (ntrogh) [inline docs/languages/ruby.md:65]: Can you add a title for accessibility purposes?

File: docs/languages/ruby.md [hunk 3]
```diff
@@ -134,17 +137,26 @@ In addition to Quick Fixes, the Ruby LSP also provides refactor options through
...
 ![Refactor extract to variable](images/ruby/ruby_lsp_refactor.png)
 
-## Debugging support with rdbg
+## Debugging
+
+The Ruby LSP extension supports debugging using the debug gem (Ruby's official debugger). Alternatively, developers can
+also install the [VS Code RDBG](https://marketplace.visualstudio.com/items?itemName=KoichiSasada.vscode-rdbg) extension
+to get debugging functionality.
+
+The following documentation is for the Ruby LSP's debugger client. Please refer to RDBG's README for instructions on how
```
Comment (ntrogh) [inline docs/languages/ruby.md:146]: Suggest adding a link to the README.

File: docs/languages/ruby.md [hunk 4]
```diff
@@ -134,17 +137,26 @@ In addition to Quick Fixes, the Ruby LSP also provides refactor options through
...
+to configure it.
 
-For debug support inside of VS Code, the Ruby LSP requires the VS Code RDBG extension to connect to debug (Ruby's official debugger).
+### Debugging tests
 
-### Install debugging support
+The Ruby LSP adds code lens buttons on top of unit tests that allow you to run examples in the test explorer, run them
```
Comment (ntrogh) [inline docs/languages/ruby.md:151]:
```suggestion
The Ruby LSP adds CodeLens buttons on top of unit tests that enable you to run examples in the Test Explorer, run them
```

**Review by ntrogh — COMMENTED** [review]
@vinistock Appreciate you refreshing the Ruby docs! I left a couple of minor suggestions.

**Comment by Saschaym** [comment]
Shopify:vs/update_ruby_docs

File: docs/languages/images/ruby/navigation.mp4
Comment (vinistock) [reply docs/languages/images/ruby/navigation.mp4:1]: Can you clarify which part exactly? Is it the toolbar?

I recorded using a window (not full screen) and on MacOS, so indeed that didn't show up, but the VS Code window should be fully contained in the video.

Comment (ntrogh) [reply docs/languages/images/ruby/navigation.mp4:1]: The VS Code title bar is cut off, which is ok when you demonstrate the editing experience, but is weird when you show the Command Palette interaction.

This is how the top of the video looks on my end:
image

**Review by ntrogh — APPROVED** [review]
Thanks for the updates, @vinistock. Would be nice if you could update the video, but shouldn't block this PR. Let me know if you want to update, otherwise good to merge.

**Review by lszomoru — DISMISSED** [review]

File: docs/languages/images/ruby/navigation.mp4
Comment (vinistock) [reply docs/languages/images/ruby/navigation.mp4:1]: Oh, I now realize what happened. I recorded the video in `.mov` first and the conversion to `.mp4` is what cut the top of the screen. I updated it so now you should get the full video.

**Review by ntrogh — APPROVED** [review]
The video looks great now, thanks @vinistock.

**Pull request merged**